operator/v1/types_network.go

```diff
@@ -0,0 +1,179 @@
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NetworkSpec   `json:"spec,omitempty"`
+	Status NetworkStatus `json:"status,omitempty"`
+}
+
+// NetworkStatus is currently unused. Instead, status
+// is reported in the Network.config.openshift.io object.
+type NetworkStatus struct {
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NetworkList contains a list of Network configurations
+type NetworkList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Network `json:"items"`
+}
+
+// NetworkSpec is the top-level network configuration object.
+type NetworkSpec struct {
+	// IP address pool to use for pod IPs.
+	// Some network providers, e.g. OpenShift SDN, support multiple ClusterNetworks.
+	// Others only support one. This is equivalent to the cluster-cidr.
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork"`
+
+	// The CIDRs to use for services
+	// Currently, all existing network providers only support a single value
+	// here, but this is an array to allow for growth.
+	ServiceNetwork []string `json:"serviceNetwork"`
+
+	// The "default" network that all pods will receive
+	DefaultNetwork DefaultNetworkDefinition `json:"defaultNetwork"`
+
+	// Additional networks to make available to pods when multiple networks
+	// are enabled.
+	AdditionalNetworks []AdditionalNetworkDefinition `json:"additionalNetworks,omitempty"`
+
+	// DisableMultiNetwork specifies whether or not multiple pod network
+	// support should be disabled. If unset, this property defaults to
+	// 'false' and multiple network support is enabled.
+	DisableMultiNetwork *bool `json:"disableMultiNetwork,omitempty"`
+
+	// DeployKubeProxy specifies whether or not a standalone kube-proxy should
+	// be deployed by the operator. Some network providers include kube-proxy
+	// or similar functionality. If unset, the plugin will attempt to select
+	// the correct value, which is false when OpenShift SDN and ovn-kubernetes are
+	// used and true otherwise.
+	// +optional
+	DeployKubeProxy *bool `json:"deployKubeProxy,omitempty"`
+
+	// KubeProxyConfig lets us configure desired proxy configuration.
+	// If not specified, sensible defaults will be chosen by OpenShift directly.
+	// Not consumed by all network providers - currently only openshift-sdn.
+	KubeProxyConfig *ProxyConfig `json:"kubeProxyConfig,omitempty"`
+}
+
+// ClusterNetwork is a subnet from which to allocate PodIPs. A network of size
+// HostPrefix (in CIDR notation) will be allocated when nodes join the cluster.
+// Not all network providers support multiple ClusterNetworks
+type ClusterNetworkEntry struct {
+	CIDR       string `json:"cidr"`
+	HostPrefix uint32 `json:"hostPrefix"`
+}
+
+// NetworkDefinition represents a single network plugin's configuration.
+// Kind must be specified, along with exactly one "Config" that matches
+// the kind.
+type DefaultNetworkDefinition struct {
```

Comment:
NetworkDefinition

Reply:
> NetworkDefinition

I'm willing to give this to @squeed if he wants it.  The straight move here will allow him to easily make his switch.

Reply:
I can live with hits, but make type and its doc consistent.